crates/valence_command/src/nodes.rs

```diff
@@ -0,0 +1,420 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+use bevy_ecs::prelude::{Component, DetectChanges, Entity};
+use bevy_ecs::query::{Added, Changed, Has, Or, With};
+use bevy_ecs::removal_detection::RemovedComponents;
+use bevy_ecs::system::{Local, ParamSet, Query, System, SystemParam};
+use bevy_ecs::world::Ref;
+use rustc_hash::{FxHashMap, FxHashSet};
+use smallvec::SmallVec;
+use valence_client::Client;
+use valence_core::__private::VarInt;
+use valence_core::protocol::encode::WritePacket;
+use valence_core::protocol::{Encode, Packet};
+
+use crate::command::CommandArguments;
+use crate::parse::{Parse, ParseObject, ParseWithData};
+use crate::pkt::{self, RawCommandTreeS2c};
+use crate::suggestions::RawParseSuggestions;
+
+pub(crate) type PCRelationVec = SmallVec<[Entity; 2]>;
```

Comment:
Does this ever exceed 2 things? If not why not just use a tuple or array?

Reply:
It is a vec for parents and children relation, so yeah it can exceed 2 elements

Reply:
Generally when you have a graph like this you want to use edges between the nodes. How do you Tel if an item in this list is parent or child?